src/App.tsx

```diff
@@ -1,39 +1,78 @@
 import React from 'react';
 import './App.scss';
+import { Clock } from './Clock';
 
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+export class App extends React.Component {
+  state = {
+    today: new Date(),
+    clockName: 'Clock-0',
+    hasClock: true,
+    timeInterval: 0,
+    nameInterval: 0,
+  };
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+  componentDidMount() {
+    document.addEventListener('click', this.addClock);
+    document.addEventListener('contextmenu', this.removeClock);
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+    this.setState({
+      timeInterval: window.setInterval(this.dateHandler, 1000),
+      nameInterval: window.setInterval(this.nameHandler, 3300),
```

Comment:
Here leave only nameInterval and timeInterval move to clock
![image](https://user-images.githubusercontent.com/70009205/231838095-cc7b87a4-008a-430e-ae91-2263b63f183d.png)